tests/storage/iso/test_iso_sr_nfs.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+import logging
+from conftest import copy_tools_iso_to_iso_sr, iso_mount_and_read_test_from_vm, clean_iso_sr
+
+class TestISOSRNFSCreateDestroy:
+    """
+    Tests that do not use fixtures that setup the SR or import VMs,
+    because they precisely need to test SR creation and destruction
+    """
+
+    def test_create_sr_nfs_with_bad_location(self, host):
+        sr = None
+        try:
+            device_config = {
+                'location': '192.168.1.1:/this/does/not/exist',
+            }
+            # Create and destroy tested in the same test to leave the host as unchanged as possible
+            host.sr_create('iso', "ISO-NFS-SR", device_config, verify=True)
+        except Exception:
+            logging.info("SR creation failed, as expected.")
+        if sr is not None:
+            sr.forget()
+            assert False, "SR creation should not have succeeded!"
+
+    def test_create_and_destroy_sr(self, host):
+        device_config = {
+            'location': '192.168.1.1:/mnt/xen/qa-iso',
+        }
+        sr = host.sr_create('iso', "ISO-NFS-SR", device_config, shared=True, verify=True)
+        sr.forget()
+
+
+@pytest.mark.usefixtures("nfs_iso_sr")
+class TestISOSRnfs():
+    """
+    Test that use fixtures to setup the SR and import VMs,
+    We Copy the tools iso on the sr, then mount it inside the VM to validate we can access it.
+    """
+
+    def test_iso_nfs(self, host, nfs_iso_sr, running_vm):
+        # create the iso sr on nfs
```

Comment:
It would be good to have ISO, SR, NFS, SMB, CIFS in uppercase in every comment and string. Function and variable names are something else and can be lowercase if relevant.